### ISSUE TITLE
Elliptic points wrong

### ISSUE THREAD

**Issue by jvoight** [assaferan/ModFrmGL2#19]
Probably a bug in FareySymbol?

```
> AttachSpec("ModFrmGL2.spec");
> 
> H := sub<GL(2,Integers(2)) | [[0,1],[1,1]]>;
> Gamma := PSL2Subgroup(H);
> EllipticPoints(Gamma);
[
    -1/2 + (1/2)*root(-3),
    1/2 + (1/6)*root(-3),
    3/2 + (1/2)*root(-3)
]
```

This contradicts the genus formula: https://beta.lmfdb.org/ModularCurve/Q/2.2.0.a.1/

**Comment by jvoight** [comment]
Possibly related, somehow the index is changing when cusps are computed?  

It must be setting the index, then recomputing it with Farey symbols to see that the group is in fact smaller than thought.  My guess is that this has to do with an assumption made somewhere about surjectivity of determinant, so the lift from H to Gamma is wrong (missing elements).

Both were found by Calvin George.

```
> AttachSpec("ModFrmGL2.spec");
> gens1 := 
> [
>     [ 3, 0, 0, 3 ],
>     [ 0, 3, 1, 3 ],
>     [ 3, 3, 2, 1 ]
> ];
> 
> gens2 := 
> [
>     [ 0, 1, 1, 0 ],
>     [ 1, 3, 2, 3 ]
> ];
> 
> H1 := sub<SL(2,Integers(4)) | gens1>;
> H2 := sub<GL(2,Integers(4)) | gens2>;
> IsConjugate(SL(2,Integers(4)), H1, H2 meet SL(2,Integers(4)));
true 
[3 0]
[3 3]
> 
> Gamma1 := PSL2Subgroup(H1);
> Gamma2 := PSL2Subgroup(H2);
> Index(Gamma1);
4
> Index(Gamma2);
4
> Cusps(Gamma1);
[
    oo
]
> Index(Gamma1);
12
```

**Comment by jvoight** [comment]
Alas, I think the error might be buried in the FareySymbols, or at least our interface to it.  

```
> AttachSpec("ModFrmGL2.spec");
> gens := [[2,1,1,1],[-1,0,2,-1],[3,-2,2,-1],[-1,0,0,-1]]; 
> Gamma := SubgroupFromGens(Gamma0(1), [Matrix(Integers(),2,2,g) : g in gens],
> 2, false);
> FareySymbol(Gamma); 
[Intrinsic procedure Print(X::SymFry) or Print(X::SymFry, level::MonStgElt) not 
present for type SymFry]
> Index(Gamma);
6
```

(The second two elements of gens generate Gamma(2), so we should be getting something of index 2 from the lift of the first element.  I threw in -1 just for safety, but it shouldn't matter.)

**Comment by jvoight** [comment]
In case it helps, here are a list of levels N and generators mod N given as eltseqs for subgroups of SL(2,N) where something goes wrong in the genus formula.

```
[ [* 2, 2,
    [
        [ 0, 1, 1, 1 ]
    ]
*], [* 4, 4,
    [
        [ 3, 0, 0, 3 ],
        [ 0, 3, 1, 3 ],
        [ 3, 3, 2, 1 ]
    ]
*], [* 4, 2,
    [
        [ 3, 0, 2, 3 ],
        [ 1, 2, 2, 1 ],
        [ 2, 3, 3, 1 ],
        [ 1, 2, 0, 1 ]
    ]
*], [* 8, 8,
    [
        [ 7, 0, 0, 7 ],
        [ 5, 4, 0, 5 ],
        [ 0, 7, 1, 7 ],
        [ 5, 0, 4, 5 ],
        [ 7, 7, 2, 1 ]
    ]
*], [* 8, 8,
    [
        [ 7, 0, 0, 7 ],
        [ 3, 3, 2, 5 ],
        [ 5, 4, 0, 5 ],
        [ 0, 7, 1, 7 ],
        [ 5, 0, 4, 5 ]
    ]
*], [* 8, 4,
    [
        [ 5, 0, 0, 5 ],
        [ 0, 3, 5, 3 ],
        [ 1, 4, 0, 1 ],
        [ 3, 0, 0, 3 ],
        [ 7, 7, 2, 1 ],
        [ 1, 0, 4, 1 ]
    ]
*], [* 8, 2,
    [
        [ 5, 6, 2, 1 ],
        [ 1, 2, 0, 1 ],
        [ 7, 0, 2, 7 ],
        [ 5, 0, 0, 5 ],
        [ 2, 7, 7, 1 ],
        [ 1, 4, 0, 1 ],
        [ 1, 0, 4, 1 ]
    ]
*], [* 9, 27,
    [
        [ 2, 5, 8, 7 ],
        [ 8, 0, 0, 8 ],
        [ 7, 7, 6, 1 ],
        [ 6, 1, 8, 3 ]
    ]
*] ]
```

**Comment by jvoight** [comment]
Finally, here's code which computes the number of elliptic points and cusps just using group theory, no FareySymbols.  Is that's what's buried somewhere in the ModularCurves repo?

```
ind := Index(G,H);

GHT := CosetTable(G,H);
pi := CosetTableToRepresentation(G,GHT);
S := Matrix([[0,-1],[1,0]]);
T := Matrix([[1,1],[0,1]]);
ST := S*T;
phiScyc := CycleStructure(pi(G!S));  // either [<2, i>, <1, j>] or [<2, i>]
if phiScyc[#phiScyc][1] eq 1 then
  nu_2 := phiScyc[#phiScyc][2];
else
  nu_2 := 0;
end if;
phiSTcyc := CycleStructure(pi(G!ST));  // either [<3, i>, <1, j>] or [<3, i>]
if phiSTcyc[#phiSTcyc][1] eq 1 then
  nu_3 := phiSTcyc[#phiSTcyc][2];
else
  nu_3 := 0;
end if;
eoo := #CycleDecomposition(pi(G!T));
```

**Comment by assaferan** [comment]
Thanks, this issue should be resolved now in pull request #22
